src/main/java/me/arminb/sara/controller/QuestionController.java

```diff
@@ -0,0 +1,107 @@
+package me.arminb.sara.controller;
+
+import me.arminb.sara.controller.models.QuestionRequest;
+import me.arminb.sara.dao.DataAccessException;
+import me.arminb.sara.entities.Question;
+import me.arminb.sara.services.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping
+public class QuestionController {
+
+    @Autowired
+    private QuestionService questionService;
+
+    @RequestMapping(value="/questions", method = RequestMethod.GET)
+    public ResponseEntity<List<Question>> findAll(@RequestParam(value="page", required=false) Integer pageNumber,
+                                              @RequestParam(value="pageCount", required = false) Integer pageCount) {
+        try {
+            List<Question> questions = questionService.findAll(pageNumber, pageCount);
+
+            return new ResponseEntity<List<Question>>(questions, HttpStatus.OK);
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<List<Question>>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value="/questions/{id}", method = RequestMethod.GET)
+    public ResponseEntity<Question> findById(@PathVariable("id") String id) {
+        try {
+            Question question = questionService.findById(id);
+            if(question != null) {
+                return new ResponseEntity<Question>(question, HttpStatus.OK);
+            } else {
+                return new ResponseEntity<Question>(HttpStatus.NOT_FOUND);
+            }
+        }
+        catch (DataAccessException e){
+            return new ResponseEntity<Question>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value="/questions/search",method = RequestMethod.GET)
```

Comment:
I would merge this with GET /questions (no search query means return all) and create just a GET /questions mapping which can be filtered with different search queries. I would do the same thing for user search in UserController.